server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DatabaseChangeController.java

```diff
@@ -47,46 +54,70 @@
             notes = "query database changing order template's detail by id")
     @GetMapping("/changingorder/templates/{id:[\\d]+}")
     public SuccessResponse<QueryDatabaseChangeChangingOrderResp> queryDatabaseChangingOrderTemplateById(
             @PathVariable Long id) {
-        return Responses.success(databaseChangeChangingOrderTemplateService.queryDatabaseChangingOrderTemplateById(id));
+        return Responses.success(templateService.detail(id));
     }
 
     @ApiOperation(value = "listDatabaseChangingOrderTemplates",
             notes = "get a list of database changing order templates")
     @GetMapping("/changingorder/templates")
-    public PaginatedResponse<QueryDatabaseChangeChangingOrderResp> listDatabaseChangingOrderTemplates(
-            @PageableDefault(size = Integer.MAX_VALUE, sort = {"id"}, direction = Direction.DESC) Pageable pageable) {
+    public PaginatedResponse<DatabaseChangeChangingOrderTemplateEntity> listDatabaseChangingOrderTemplates(
+            @PageableDefault(size = Integer.MAX_VALUE, sort = {"id"}, direction = Direction.DESC) Pageable pageable,
+            @RequestParam(required = false, name = "name") String name,
+            @RequestParam(required = false, name = "creatorId") Long creatorId,
+            @RequestParam(required = true, name = "projectId") Long projectId) {
+        QueryDatabaseChangeChangingOrderParams queryDatabaseChangeChangingOrderParams =
+                QueryDatabaseChangeChangingOrderParams.builder()
+                        .name(name)
+                        .creatorId(creatorId)
+                        .projectId(projectId)
+                        .build();
         return Responses
-                .paginated(databaseChangeChangingOrderTemplateService.listDatabaseChangingOrderTemplates(pageable));
+                .paginated(templateService.listTemplates(pageable,
+                        queryDatabaseChangeChangingOrderParams));
     }
 
     @ApiOperation(value = "deleteDatabaseChangingOrderTemplateById",
             notes = "delete database changing order template by id")
     @DeleteMapping("/changingorder/templates/{id:[\\d]+}")
     public SuccessResponse<Boolean> deleteDatabaseChangingOrderTemplateById(@PathVariable Long id) {
         return Responses
-                .success(databaseChangeChangingOrderTemplateService.deleteDatabaseChangingOrderTemplateById(id));
+                .success(templateService.delete(id));
+    }
+
+    @ApiOperation(value = "exists", notes = "Returns whether an database changing order template exists")
+    @RequestMapping(value = "/changingorder/templates/exists", method = RequestMethod.GET)
+    public SuccessResponse<DatabaseChangingOrderTemplateExists> exists(@RequestParam String name,
+            @RequestParam Long projectId) {
+        return Responses.success(templateService.exists(name, projectId));
+    }
+
+    @ApiOperation(value = "enables", notes = "Returns whether an database changing order template enables")
```

Comment:
not nececesary, front end can reuse the get API

Reply:
removed it